lua/nvim-surround/filetype.lua

```diff
@@ -4,226 +4,166 @@ local M = {}
+            },
+        })
     end,
-}
-
-M.rust = { -- TODO: Try to add support for macros?
-    setup = function()
-        M.set_autocmd("rust", function()
-            require("nvim-surround").buffer_setup({
-                surrounds = {
-                    ["f"] = {
-                        find = function()
-                            return config.get_selection({ node = "call_expression" })
-                        end,
-                    },
+    rust = function() -- TODO: Try to add support for macros?
+        require("nvim-surround").buffer_setup({
+            surrounds = {
+                ["f"] = {
+                    find = function()
+                        return config.get_selection({ node = "call_expression" })
+                    end,
                 },
-            })
-        end)
+            },
+        })
     end,
-}
-
-M.typescriptreact = {
-    setup = function()
-        M.set_autocmd("typescriptreact", function()
-            require("nvim-surround").buffer_setup({
-                surrounds = {
-                    ["t"] = {
-                        find = function()
-                            return config.get_selection({ node = "jsx_element" })
-                        end,
-                    },
-                    ["T"] = {
-                        find = function()
-                            return config.get_selection({ node = "jsx_element" })
-                        end,
-                    },
+    typescriptreact = function()
+        require("nvim-surround").buffer_setup({
+            surrounds = {
+                ["t"] = {
+                    find = function()
+                        return config.get_selection({ node = "jsx_element" })
+                    end,
                 },
-            })
-        end)
+                ["T"] = {
+                    find = function()
+                        return config.get_selection({ node = "jsx_element" })
+                    end,
+                },
+            },
+        })
     end,
-}
-
-M.vim = {
-    setup = function()
-        M.set_autocmd("vim", function()
-            require("nvim-surround").buffer_setup({
-                surrounds = {
-                    ["f"] = {
-                        find = function()
-                            return config.get_selection({ node = "call_expression" })
-                        end,
-                    },
+    vim = function()
+        require("nvim-surround").buffer_setup({
+            surrounds = {
+                ["f"] = {
+                    find = function()
+                        return config.get_selection({ node = "call_expression" })
+                    end,
                 },
-            })
-        end)
+            },
+        })
     end,
 }
 
+M.setup = function()
+    vim.api.nvim_create_autocmd("FileType", {
+        pattern = vim.tbl_keys(M.filetype_configurations),
+        callback = function()
+            M.filetype_configurations[vim.bo.filetype]()
```

Comment:
Looks like there should be a nil-check here in case there is no function for the given filetype :)

Reply:
@smjonas I actually forgot I accounted for this, but there will *always* exist a function for the given filetype, since the autocommand uses the `M.filetype_configurations` table to even run the autocommand to begin with. So this shouldn't be an issue as far as I can tell :) Thanks for checking though!

Reply:
Ah, you're right, I missed that :D